instructionAPI/src/BinaryFunction.C

```diff
@@ -32,425 +32,375 @@
-        retVal = retVal | m_arg2->bind(expr, value);
-		
-		if(retVal) 
-			clearValue();
-    
-        return retVal;
+  }
+
+  Result doAnd(Result arg1, Result arg2, Result_Type ResultT) {
+    switch(ResultT) {
+      case s8:
+      case u8:
+        return Result(u8, arg1.convert<Result_type2type<u8>::type>() &
+                              arg2.convert<Result_type2type<u8>::type>());
+      case s16:
+      case u16:
+        return Result(u16, arg1.convert<Result_type2type<u16>::type>() &
+                               arg2.convert<Result_type2type<u16>::type>());
+      case s32:
+      case u32:
+        return Result(u32, arg1.convert<Result_type2type<u32>::type>() &
+                               arg2.convert<Result_type2type<u32>::type>());
+      case s48:
+      case u48:
+        return Result(u48, arg1.convert<Result_type2type<u48>::type>() &
+                               arg2.convert<Result_type2type<u48>::type>());
+      case s64:
+      case u64:
+        return Result(u64, arg1.convert<Result_type2type<u64>::type>() &
+                               arg2.convert<Result_type2type<u64>::type>());
+      /*case sp_float:
+                    return Result(ResultT, arg1.convert<Result_type2type<sp_float>::type >() +
+      arg2.convert< Result_type2type<sp_float>::type >()); case dp_float: return Result(ResultT,
+      arg1.convert<Result_type2type<dp_float>::type >() + arg2.convert<
+                      Result_type2type<dp_float>::type >());*/
+      case bit_flag:
+        return Result(ResultT, arg1.convert<Result_type2type<bit_flag>::type>() &
+                                   arg2.convert<Result_type2type<bit_flag>::type>());
+      default:
+        // m512 and dbl128 not implemented yet...
+        return Result(ResultT);
     }
+  }
 
-    void BinaryFunction::apply(Visitor* v)
-    {
-        m_arg1->apply(v);
-        m_arg2->apply(v);
-        v->visit(this);
+  Result doOr(Result arg1, Result arg2, Result_Type ResultT) {
+    switch(ResultT) {
+      case s8:
+      case u8:
+        return Result(u8, arg1.convert<Result_type2type<u8>::type>() |
+                              arg2.convert<Result_type2type<u8>::type>());
+      case s16:
+      case u16:
+        return Result(u16, arg1.convert<Result_type2type<u16>::type>() |
+                               arg2.convert<Result_type2type<u16>::type>());
+      case s32:
+      case u32:
+        return Result(u32, arg1.convert<Result_type2type<u32>::type>() |
+                               arg2.convert<Result_type2type<u32>::type>());
+      case s48:
+      case u48:
+        return Result(u48, arg1.convert<Result_type2type<u48>::type>() |
+                               arg2.convert<Result_type2type<u48>::type>());
+      case s64:
+      case u64:
+        return Result(u64, arg1.convert<Result_type2type<u64>::type>() |
+                               arg2.convert<Result_type2type<u64>::type>());
+      /*case sp_float:
```

Comment:
This comment is kind of hard to read.

Reply:
That looks like dead code. I'll remove it and the ones above.